### PR TITLE
➕✨ Add socket.io and Lobbies page functions

### LINKED ISSUE #4: Add lobby-page functionality
## User story
__As a__:  User
__I want to__:  select a lobby to join from a list of lobbies or create a lobby
__So I can__:  start playing a game with other players

## Acceptance criteria

- [x] When pressing 'Create Game' the player should be redirected to the game board
- [x] When pressing 'Create Game' the created game should show up in the lobbies list, so other players can join

### PR DESCRIPTION
**Pull request by fre-ben**
- Add socket.io
- Add Lobbies Page functionality including sockets

Clicking the "Create Game" button now adds a LobbyListItem to the DOM and redirects to the created lobby.
Also other players connecting to the lobby will be able to see those created lobbies and are able to join them.

Sorry for the long list of changed files, while trying to figure out socket.io and my backend I forgot to make clear and constant commits 🦝 
Also at the start of this PR I didn't know where I was heading with it.

Deployment: https://sisu-pipelin-socket-hylscdik7y.herokuapp.com/
Fixes #4

### CODE REVIEW

File: server/lib/games.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { Game, Player, Card } from "./gameTypes";
```
Comment (phlgr) [inline server/lib/games.ts:1]: You should remove unnecessary imports

**Review by phlgr — APPROVED** [review]
GJ 🚀

File: server/lib/games.ts (hunk 1 above)
Comment (fre-ben) [reply server/lib/games.ts:1]: Yup, thanks!

**Review by mewiemers — COMMENTED** [review]
LGTM!🎯

**Pull request merged**